specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/queue.json

```diff
@@ -0,0 +1,591 @@
+            "$ref": "#/parameters/QueueName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- returned table with the specified table name successfully.",
+            "schema": {
+              "$ref": "#/definitions/StorageQueue"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "QueueService"
+        ],
+        "operationId": "DeleteQueue",
+        "x-ms-examples": {
+          "QueueOperationDelete": {
+            "$ref": "./examples/QueueOperationDelete.json"
+          }
+        },
+        "description": "Deletes the queue with the specified queue name, under the specified account if it exists.",
+        "parameters": [
+          {
+            "$ref": "./storage.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "./storage.json#/parameters/StorageAccountName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/QueueName"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "No Content - Successfully deleted the queue with the specified queue name"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/queueServices/default/queues": {
+      "get": {
+        "tags": [
+          "QueueService"
+        ],
+        "operationId": "ListQueue",
+        "x-ms-examples": {
+          "QueueOperationList": {
+            "$ref": "./examples/QueueOperationList.json"
+          }
+        },
+        "description": "Gets a list of all the queues under the specified storage account",
+        "parameters": [
+          {
+            "$ref": "./storage.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "./storage.json#/parameters/StorageAccountName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "$maxpagesize",
```

Comment:
>$maxpagesize [](start = 21, length = 12)

This should be $top. (Take a look at [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource) under "Paging Response Body".)

Reply:
Thanks for your review,
We have existing APIs which make use of $maxpagesize parameter, I have followed the same pattern here.  example: https://docs.microsoft.com/en-us/rest/api/storagerp/fileshares/list

Moreover the server-side has already rolled out with this parameter.

Reply:
I guess we'll have to allow it.

---
In reply to: [422474575](https://github.com/Azure/azure-rest-api-specs/pull/9259#discussion_r422474575) [](ancestors = 422474575)